app/server/lib/DocStorage.ts

```diff
@@ -1465,7 +1465,18 @@ export class DocStorage implements ISQLiteDB, OnDemandStorage {
         WHERE a.value = ${attId}`);
       }
     }
-    return (await this.all(queries.join(' UNION ALL '))) as any[];
+    try {
+      return (await this.all(queries.join(' UNION ALL '))) as any[];
+    }
+    catch (e) {
+      // We throw an informative error if we fail to process the attachment references
+      // cf: https://github.com/gristlabs/grist-core/issues/1565
+      const errorMessage = `findAttachmentReferences failed: unable to process attachment references` +
+      `for users with complicated access rules. ` +
+      `Make sure that there are no errors in attachments columns across the document. Details: ${e.message}`;
```

Comment:
So the problem if I understand it is that a call to `json_each` failing. I see in nearby functions some checking for `json_valid` first. Would it be more consistent to also do that in this query? And then the process could proceed rather than failing.

I understand the point about not investing so much in this corner area though.

Reply:
You are right, it was an easy fix so I did it :)

Also, I guess it's always better to not pass invalid (and untrusted) user input into a sql query 🙃